cli/azd/pkg/project/container_helper.go

```diff
@@ -473,6 +484,41 @@ func (ch *ContainerHelper) runRemoteBuild(
 	return imageName, nil
 }
 
+// runDotnetPublish builds and publishes the container image using `dotnet publish`. It returns the full remote image name.
+func (ch *ContainerHelper) runDotnetPublish(
+	ctx context.Context,
+	serviceConfig *ServiceConfig,
+	target *environment.TargetResource,
+	progress *async.Progress[ServiceProgress],
+) (string, error) {
+	progress.SetProgress(NewServiceProgress("Logging into registry"))
+
+	dockerCreds, err := ch.Credentials(ctx, serviceConfig, target)
+	if err != nil {
+		return "", fmt.Errorf("logging in to registry: %w", err)
+	}
+
+	progress.SetProgress(NewServiceProgress("Publishing container image"))
+
+	imageName := fmt.Sprintf("%s:%s",
+		ch.DefaultImageName(serviceConfig),
+		ch.DefaultImageTag())
+
+	_, err = ch.dotNetCli.PublishContainer(
```

Comment:
It's really difficult to see the forest for the trees right now given the multiple, multiple layers we're stacking on top of here, and our azd product stretch from compile-time to runtime --

But I think this does end up changing the default listening port from 80 to 8080 in the final image. Maybe we could consider updating the generated infra assets from easyInit/compose with this idea in mind for .NET languages:

- add_configure_host.go, L240
- infra_confirm.go, L215

I was reminded of this by a very timely community issue feedback #4583

Reply:
I did this and verified that the in the easy-init flow things worked as expected.  I assume they will also work for the compose flow where `infra` is not generated, but I didn't explicitly test it.

It made me smile to see the whole thing work end to end, and it was nice how little I had to change to make it all work!